SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp

```diff
@@ -857,6 +865,44 @@ FString FSpatialGDKEditorToolbarModule::GetOptionalExposedRuntimeIP() const
 	}
 }
 
+void FSpatialGDKEditorToolbarModule::UpdateMobileClient() const
+{
+	const USpatialGDKEditorSettings* SpatialGDKSettings = GetDefault<USpatialGDKEditorSettings>();
+	const UIOSRuntimeSettings* IOSRuntimeSettings = GetDefault<UIOSRuntimeSettings>();
+	const FString ProjectName = FApp::GetProjectName();
+
+	const FString CommandLineArgs = FString::Printf(TEXT( "../../../%s/%s.uproject %s -game -log -workerType UnrealClient"), *ProjectName, *ProjectName, *(SpatialGDKSettings->ExposedRuntimeIP));
```

Comment:
`../../../` terrifies me.

Reply:
I think there are helper methods for getting the `.uproject`.

Reply:
>../../../ terrifies me.

me too, but this seems to be the only path that works for iOS.

Reply:
had a look, but couldn't find a helper method for getting the uproject

Reply:
I couldn't find a helper either but I did see this pattern around the engine.
```
FString ProjectPath = FPaths::IsProjectFilePathSet() ? FPaths::ConvertRelativePathToFull(FPaths::GetProjectFilePath()) : FPaths::RootDir() / FApp::GetProjectName() / FApp::GetProjectName() + TEXT(".uproject");
```

Reply:
that won't work here because this will result in the path on the Mac machine, while we need the path that it will have on the iOS device. looking into whether we can get this path somehow

Reply:
ok, I spent the last 2 days digging through Unreal trying to figure out a way on how to identify the correct way to construct the project path for mobile. They basically hardcode it...

https://github.com/improbableio/UnrealEngine/blob/4.22-SpatialOSUnrealGDK-release/Engine/Source/Programs/AutomationTool/AutomationUtils/DeploymentContext.cs#L408

```
public const string UProjectCommandLineArgInternalRoot = "../../../";
UProjectCommandLineArgInternalRoot + RelativeProjectRootForStage.Name + "/" + ShortProjectName + ".uproject"
```

I also posted in UDN if there is a better way to do it, but for now it looks like we have to keep it hardcoded

Reply:
Nice job posting, we'll use the current solution for now and update if possible.
Could you please add a comment in the code indicating this is a work-around for getting the path on iOS